src/assets/TokensController.test.ts

```diff
@@ -27,13 +29,13 @@ describe('TokensController', () => {
       },
     });
 
-    sinon
+    instEthProvStub = sinon
       .stub(tokensController, '_instantiateNewEthersProvider')
       .callsFake(() => null);
   });
 
   afterEach(() => {
-    sinon.restore();
+    instEthProvStub.restore();
```

Comment:
What do you think about continuing to use Sinon for mocks and keeping `sinon.restore()` here?

Reply:
my 2c:

1. we are already using jest, which comes with stub/mocks. Using the mocks that come with the testing framework being used is what most developers would expect.
2. our [jest config uses clear mocks](https://github.com/MetaMask/controllers/blob/main/jest.config.js#L26), which works great when using jest mocks
3. jest has wider support, making very tricky to test situations a lot easier via module-level mocking.
4. we are already have a jest  __mocks__ which indicated to me that we are using jest mocks. intermixing of sinon and jest mocks is not super fun. 

I do like sinon, always have. But when using jest as a testing framework, using sinon just feels a bit stubborn. 

TLDR; We want to use sinon only. I'd argue we should use jest only. Happy to disagree and commit to using sinon.

Reply:
Pushed a commit that reverts to using sinon / sinon.restore

Reply:
Related: https://github.com/MetaMask/controllers/issues/744
I'm all for keeping PRs simple though, so changing that separately might be preferable anyway.

Reply:
@BelfordZ Okay thanks. As Mark pointed out moving everything to Jest mocks in this repo is something we definitely we want to do, and I agree with all your points about using Jest over Sinon. The thing I forgot to say here is that I think from a maintainability perspective, it would be better to convert all tests in a file to use Jest mocks in one go rather than having some tests use Jest mocks and some use Sinon mocks. The current situation is definitely confusing but I feel like that would help control the chaos slightly better. So that's where my suggestion came from. That said if you feel like taking on converting a file we would definitely appreciate it :)